Inwentaryzacja/Inwentaryzacja/Inwentaryzacja/views/view-allReports/ReportDetailsView.xaml.cs

```diff
@@ -17,14 +19,91 @@ namespace Inwentaryzacja.views.view_allReports
+            reportId = id;
             FillViewWithText(headerText, roomText, createDate, createTime, ownerText, inThisRoom, moveToRoom, moveFromRoom, inAnotherRoom, scannedAll, scannedAllLabel, moveFromRoomLabel, moveToRoomLabel, inAnotherRoomLabel, inThisRoomLabel);
-            DetailsButtonsImplementation(scannedAllDetails, inThisRoomDetails, movedToRoomDetails, movedFromRoomDetails, inAnotherRoomDetails);            
+            DetailsButtonsImplementation(scannedAllDetails, inThisRoomDetails, movedToRoomDetails, movedFromRoomDetails, inAnotherRoomDetails);
+            CrossDownloadManager.Current.CollectionChanged += (sender, e) =>
+	            System.Diagnostics.Debug.WriteLine(
+		            "[DownloadManager] " + e.Action +
+                    " -> New items: " + (e.NewItems?.Count ?? 0) +
+                    " at " + e.NewStartingIndex + 
+                    " || Old items: " + (e.OldItems?.Count ?? 0) + 
+                    " at " + e.OldStartingIndex
+	            );
+        }
+
+        /// <summary>
+        /// Funkcja odpowiadajaca za pobranie pliku
+        /// </summary>
+        public async void DownloadFile(String FileName)
+        {
+	        await Task.Yield();
+	        //await Navigation.PushPopupAsync(new DownLoadingPage());
+	        await Task.Run(() =>
+	        {
+		        var downloadManager = CrossDownloadManager.Current;
+		        var file = downloadManager.CreateDownloadFile(FileName);
+                downloadManager.Start(file, true);
+
+                while (isDownloading)
+                {
+	                isDownloading = IsDownloading(file);
+                }
+	        });
+
+	        if (!isDownloading)
+	        {
+		        await DisplayAlert("Status pliku", "Raport pobrany pomyślnie", "OK");
+	        }
+        }
+
+        /// <summary>
+        /// Funkcja odpowiadajaca za sprawdzenie czy plik jest pobierany
+        /// </summary>
+        public bool IsDownloading(IDownloadFile File)
+        {
+            if (File == null)
+                return false;
+
+            switch (File.Status)
+            {
+	            case DownloadFileStatus.INITIALIZED:
+	            case DownloadFileStatus.PAUSED:
+	            case DownloadFileStatus.PENDING:
+	            case DownloadFileStatus.RUNNING:
+                    return true;
+                
+                case DownloadFileStatus.COMPLETED:
+                case DownloadFileStatus.CANCELED:
+                case DownloadFileStatus.FAILED:
+	                return false;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Funkcja odpowiadajaca za pobranie pliku po kliknieciu przycisku
+        /// </summary>
+        private void DownloadBtn_Clicked(object sender, EventArgs e)
+        {
+	        var Url = "https://aplikacja-do-inwentaryzacji.000webhostapp.com/api/pdfGenerator/"+reportId+"/1";
```

Comment:
A i tutaj, zamień to wczytanie na sztywno na pobranie url z api controllera tam jest ścieżka, prezentacja ma być na localhost wtedy podmiana będzie w 1 miejscu.
Jak jest private to zrób getera lub public nawet i readonly